tests/test_degrade.py

```diff
@@ -537,6 +537,96 @@ def test_degrade_widemask_wmean(self):
                               sparse_map.degrade, nside_out=nside_out, reduction='wmean',
                               weights=weights)
 
+    def test_degrade_lowres_float(self):
+        """
+        Test HealSparse.degrade in the case where the target resolution
+        is smaller than the original coverage resolution (nside_out < nside_coverage)
+        """
+
+        nside_coverage = 32
+        nside_map = 256
+        nside_out = 8
+        pxnums = np.arange(2000)
+        pxvalues = np.random.random(size=2000).astype(np.float64)
+        weights = 0.5+0.5*np.random.random(size=2000).astype(np.float64)
+        for method in ['mean', 'std', 'max', 'mean', 'median', 'wmean',
+                       'sum', 'prod']:
+            if method == 'wmean':
+                wgt = weights
+            else:
+                wgt = None
+            sparse_map = healsparse.HealSparseMap.make_empty(nside_coverage, nside_map, dtype=np.float64)
+            sparse_map2 = healsparse.HealSparseMap.make_empty(nside_out, nside_map, dtype=np.float64)
+            sparse_map.update_values_pix(pxnums, pxvalues)
+            sparse_map2.update_values_pix(pxnums, pxvalues)
+            # Check that a warning is raised
+            # testing.assert_raises(ResourceWarning, sparse_map.degrade,
+            #                      nside_out, reduction=method, weights=weights)
+            with testing.assert_raises(ResourceWarning):
+                sparse_map = sparse_map.degrade(nside_out, reduction=method, weights=wgt)
```

Comment:
I'm not sure how this could work ... the weights aren't a healsparse map but https://github.com/LSSTDESC/healsparse/blob/master/healsparse/healSparseMap.py#L1503-L1504